Add preferred terms list to style guide (#386)
ansible/ansible-documentation: docs/docsite/rst/dev_guide/style_guide/preferred_terms.rst

.. _preferred_terms:

Preferred terminology
=====================

.. contents::
  :local:

The following is the list of preferred terms for Ansible ecosystem projects.

> Ansible

Reviewer: Could you make a glossary out of these terms? This would register the terms in the corresponding Sphinx domain and they'll be linkable via the `:term:` role:
```suggestion
.. glossary::

   Ansible
```
More info on this: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#glossary.

>    Use on all occurences or `ansible` if reffering to the CLI command. Preference is to use this to refer to the Ansible community package.

Reviewer: RST typically requires double backticks for highlighting.

Also: s/reffering/referring/

>    Use on all occurences or `ansible` if reffering to the CLI command. Preference is to use this to refer to the Ansible community package.

Reviewer 2: ```suggestion
   Use on all occurrences or ``ansible`` if referring to the CLI command. Preference is to use this to refer to the Ansible community package.
```

Ansible Builder

>    Use on first occurence or `ansible-builder` if referring to the CLI command. Subsequent occurences can be Builder.

Reviewer 2: ```suggestion
   Use on first occurrence or ``ansible-builder`` if referring to the CLI command. Subsequent occurrences can be Builder.
```

Ansible Collections
   Use on first occurrence. Subsequent occurrences can be collections.

> Ansible Content Development kit (CDK)

Reviewer 2: ```suggestion
Ansible Content Development Kit (CDK)
```

>    Use on first occurence. Subsequent occurences can be Ansible CDK

Reviewer 2: ```suggestion
   Use on first occurrence. Subsequent occurrences can be Ansible CDK
```

Ansible Core

>    Use on all occurences, though preference is `ansible-core` after first occurence to distiguish from the Ansible community package.

Reviewer 2: ```suggestion
   Use on all occurrences, though preference is ``ansible-core`` after first occurrence to distinguish from the Ansible community package.
```

Ansible Creator

>    Use on first occurence or `ansible-creator` if referring to the CLI command.

Reviewer 2: ```suggestion
   Use on first occurrence or ``ansible-creator`` if referring to the CLI command.
```

Ansible Execution Environment(EE)
   Use on first occurrence. Subsequent occurrences can be Execution Environment or EE.
Ansible Galaxy

>    Use on first occurence. Subsequent occurences can be Galaxy. Do not use Galaxy NG.

Reviewer 2: ```suggestion
   Use on first occurrence. Subsequent occurrences can be Galaxy. Do not use Galaxy NG.
```

Ansible Lint

>    Use on all occurrences or `ansible-lint` if referring to the CLI command

Author: ```suggestion
   Use on all occurrences or ``ansible-lint`` if referring to the CLI command
```

Ansible Molecule

>    Use on first occurrence. Subsequent occurrences can be Molecule or `molecule` if referring to the CLI command.

Author: ```suggestion
   Use on first occurrence. Subsequent occurrences can be Molecule or ``molecule`` if referring to the CLI command.
```

Ansible Navigator

>    Use first occurrence. Subsequent occurrences can be Navigator or `ansible-navigator` if referring to the CLI.

Author: ```suggestion
   Use first occurrence. Subsequent occurrences can be Navigator or ``ansible-navigator`` if referring to the CLI.
```

Ansible Playbook 
   Use on first occurrence. Subsequent occurrences can be playbook.
`ansible-pylibssh`

>    Use on all occurences.

Reviewer 2: ```suggestion
   Use on all occurrences.
```

Ansible Roles
   Use on first occurrence. Subsequent occurrences can be roles.
Ansible Rule
   Use on first occurrence. Subsequent occurrences can be rule.
Ansible Rulebook
   Use on first occurrence. Subsequent occurrences can be rulebook.
Ansible Runner

>    Use on first occurence or `ansible-runner` if referring to the CLI command. Subsequent occurences can be Runner.

Reviewer 2: ```suggestion
   Use on first occurrence or ``ansible-runner`` if referring to the CLI command. Subsequent occurrences can be Runner.
```

Ansible SDK

>    Use on all occurences. 

Reviewer 2: ```suggestion
   Use on all occurrences. 
```

Ansible Sign

>    Use on all occurences or `ansible-sign` if referring to the CLI command.

Reviewer 2: ```suggestion
   Use on all occurrences or ``ansible-sign`` if referring to the CLI command.
```

`ansible-test`

>    Use on all occurences as the CLI command.

Reviewer 2: ```suggestion
   Use on all occurrences as the CLI command.
```

Ansible VSCode Extension by Red Hat

>    Use on first occurence. Subsequent occurences can be Ansible VSCode Extension.

Reviewer 2: ```suggestion
   Use on first occurrence. Subsequent occurrences can be Ansible VSCode Extension.
```

Pinakes

>    Use on all occurences.

Reviewer 2: ```suggestion
   Use on all occurrences.
```

Receptor

>    Use on all occurrences or `receptor` if referring to the CLI command.

Author: ```suggestion
   Use on all occurrences or ``receptor`` if referring to the CLI command.
```

.. _preferred_terms:

Preferred terminology
=====================

.. contents::
  :local:

The following is the list of preferred terms for Ansible ecosystem projects.

.. glossary::

   Ansible

>    Use on all occurrences or ``ansible`` if referring to the CLI command. Preference is to use this to refer to the Ansible community package.

Reviewer: The definitions under each term should have extra indentation per https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#glossary
```suggestion
      Use on all occurrences or ``ansible`` if referring to the CLI command. Preference is to use this to refer to the Ansible community package.
```

Author: ok thanks. Frankly I was so mortified by all my spelling errors I forgot to take a closer look at how the glossary works. Thanks

Reviewer: I tend to use https://github.com/sphinx-contrib/spelling for detecting typos...

Reviewer 2: Personally I like codespell https://github.com/codespell-project/codespell

Reviewer: I didn't know it can check RST.

Author: Ok now I have a duplicate term definition problem because we already have an actual glossary in the appendix.  I'm inclined to remove the glossary from this section because it's not really defining the terms, just how to type them in documentation.

> Ansible Builder

Reviewer: May I suggest injecting empty line separators before each term? It makes a huge visual difference when looking at the source code.
```suggestion

  Ansible Builder
```

Reviewer: @samccann also, my original suggested change only indented the first term but all the others should be indented too, it just wasn't convenient to do this in a browser input.

.. _preferred_terms:

Preferred terminology
=====================

.. contents::
  :local:

The following is the list of preferred terms for Ansible ecosystem projects.

   Ansible
      Use on all occurrences or ``ansible`` if referring to the CLI command. Preference is to use this to refer to the Ansible community package.

   Ansible Builder
      Use on first occurrence or ``ansible-builder`` if referring to the CLI command. Subsequent occurrences can be Builder.

   Ansible Collections

>       Use on first occurrence. Subsequent occurrences can be collections.

Reviewer 3: Hmm, if C in collections is capitalized in Ansible Collections, it really sounds like a product to me (which implies there is one Ansible Collections thing, and not that there are many Ansible collections).

The same applies to Ansible Execution Environments, Ansible Playbook, Ansible Roles, Ansible Rule, Ansible Rulebook, i.e. basically everything where from the second occurrence the second term is used lower-case.

Author: We went back and forth (and back again) on this one.  I think the request stems from https://github.com/ansible/ansible-documentation/issues/152#issuecomment-1660625722

So I tried to split it in the middle-keep the caps for the first full occurrence, and then drop later on for most, unless it's the official project name.

Author: Rereading https://github.com/ansible/ansible-documentation/issues/152#issuecomment-1661862668 and will try to apply in an update here. That comment did suggest Ansible collections, not Ansible Collections.

>    Ansible collections
>       Use on first occurrence. Subsequent occurrences can be collections.

Reviewer 3: Same here:
```suggestion
   Ansible collection
      Use on first occurrence. Subsequent occurrences can be collection.
```

   Ansible Content Development Kit (CDK)
      Use on first occurrence. Subsequent occurrences can be Ansible CDK.

   Ansible Core
      Use on all occurrences, though preference is ``ansible-core`` after first occurrence to distinguish from the Ansible community package.

   Ansible Creator
      Use on first occurrence or ``ansible-creator`` if referring to the CLI command.

   Ansible Execution Environment(EE)
      Use on first occurrence. Subsequent occurrences can be Execution Environment or EE.

   Ansible Galaxy
      Use on first occurrence. Subsequent occurrences can be Galaxy. Do not use Galaxy NG.

   Ansible Lint
      Use on all occurrences or ``ansible-lint`` if referring to the CLI command.

   Ansible Molecule
      Use on first occurrence. Subsequent occurrences can be Molecule or ``molecule`` if referring to the CLI command.

   Ansible Navigator
      Use first occurrence. Subsequent occurrences can be Navigator or ``ansible-navigator`` if referring to the CLI.

   Ansible playbook 
      Use on first occurrence. Subsequent occurrences can be playbook.

   `ansible-pylibssh`
      Use on all occurrences.

>    Ansible roles
>       Use on first occurrence. Subsequent occurrences can be roles.

Reviewer 3: Since the other terms (rule, rulebook, playbook, execution environment) are also singular, why not this one?
```suggestion
   Ansible role
      Use on first occurrence. Subsequent occurrences can be role.
```

   Ansible rule
      Use on first occurrence. Subsequent occurrences can be rule.

   Ansible rulebook
      Use on first occurrence. Subsequent occurrences can be rulebook.

   Ansible Runner
      Use on first occurrence or ``ansible-runner`` if referring to the CLI command. Subsequent occurrences can be Runner.

   Ansible SDK
      Use on all occurrences. 

   Ansible Sign
      Use on all occurrences or ``ansible-sign`` if referring to the CLI command.

>    `ansible-test`

Reviewer 3: ```suggestion
   ``ansible-test``
```
